### PR TITLE
APT: Run debootstrap with --variant=minbase

### PR DESCRIPTION
**Pull request by michal42**
This selects only packages with Priority: required, which is more in
line with what the other package managers are doing. A patch for
kiwi-descriptions has been submitted to adapt to this new behavior.

### CODE REVIEW

**Comment by michal42** [comment]
https://github.com/SUSE/kiwi-descriptions/pull/17 is the kiwi-descriptions fixup

**Comment by schaefi** [comment]
I like the change to make "onlyRequired" the default behavior and the code change is good

What came to my mind is; In the packages section we support the attribute patternType="onlyRequired" | "plusRecommended", I think it would be great if this patch could be extended in a way which allows the selection of more than Priority: required packages if the patternType is set to another value so that we don't loose the possibility we had before

Thoughts ?

**Comment by schaefi** [comment]
I was thinking on mapping patternType="plusRecommended" to e.g scratchbox variant or none variant. But I'm not sure if that would be ok. Alternatively patternType could be extended to allow the variant types and if not set no variant will be set

**Comment by davidcassany** [comment]
I like the idea to mimic somehow the patternType="plusRecommended" to some or none variant and patternType="onlyRequired" to minbase variant.

For plusRecommended, I think I prefer to leave it without any variant and just use defaults. I guess that having the minbase variant as default is enough, since you are always capable to easily add packages to your description; but is hard to remove packages that have been already included without being nasty. 

Also not sure if fakechroot is smaller than minbase, does anyone have a clue? fakechroot sounds also like a good candidate for the default bootstrap variant.

In short, my opinion would be to leave as a default (onlyRequired) the most minimalist option and leave it without any variant for plusRecommended. We could also plan to support some way to set the variant from the description file and support all of them,; but I would not go that way, I don't think there is a need.

**Comment by michal42** [comment]
Thanks for the review. I never used plusRecommended, but I can have a look.

**Comment by schaefi** [comment]
@michal42 just to put me right, you still plan to update your code with the suggested changes right ?

Thanks

**Comment by michal42** [comment]
Sorry for the delay. Yes, that has been the plan. I pushed the change now, let me know what you think and enjoy your weekend.

Michal

**Comment by schaefi** [comment]
yeah this looks great to me :+1:
